astropy/coordinates/angles/core.py

```diff
@@ -306,8 +306,16 @@ def to_string(
             will be an array with a unicode dtype.
 
         """
+        if decimal and sep in ("dms", "hms"):
```

Comment:
@dhomeier I think this implements your suggestion. To me it feels like a `TypeError` would be more appropriate here but using a `ValueError` makes it so that the error type is the same as with older versions (just the message is different), which makes it easier to catch consistently downstream.

Reply:
It's either one of the inputs to `decimal` or `sep` having an illegal value, so `ValueError` seems quite appropriate to me. In fact I think this could check for `sep != "fromunit"` as well and then completely replace the second test below (currently any other string passed in `sep` will still slip through to that point).
I was also thinking whether keeping the same error message for consistency would be better, but actually the `f"'{unit}' can not be represented in sexagesimal notation"` in the second check seems misleading at best – `sep` simply does not make any sense for non-sexagesimal notation, and `"fromunit"` just masks that it is the default value, which is simply ignored with `decimal` format.

Reply:
OK, had to read through my own and other's comments from 3 years ago to realise this was all rather more complex.
But the case I had in mind, which I think would still be reproduced with the current version, is
```python
>>> a = Angle(np.pi/12, 'rad')
>>> a.to_string(sep=':', decimal=True)
ValueError: 'rad' can not be represented in sexagesimal notation
>>> d = Angle(15, 'deg')
>>> d.to_string(sep=':', decimal=True)
ValueError: 'deg' can not be represented in sexagesimal notation
```
The first message is arguably correct, but does not cover the full problem, while the second just doesn't make sense.

Reply:
Oh, nice catch. I confirm that your examples are unchanged in the current state of this PR. I'll try to fix this too here !

Reply:
I adjusted the new test to include these examples. Hopefully the new message makes more sense.